src/libs/collect.c

```diff
@@ -965,13 +1038,14 @@ static void tree_view(dt_lib_collect_rule_t *dr)
             GtkTreeIter iter;
 
             pth = dt_util_dstrcat(pth, format_separator, *token);
+            if(times && !*(token + 1)) pth[10] = ' ';
 
             gchar *pth2 = g_strdup(pth);
             pth2[strlen(pth2) - 1] = '\0';
-
             gtk_tree_store_insert(GTK_TREE_STORE(model), &iter, common_length > 0 ? &parent : NULL, insert_position);
             gtk_tree_store_set(GTK_TREE_STORE(model), &iter, DT_LIB_COLLECT_COL_TEXT, *token,
-                               DT_LIB_COLLECT_COL_PATH, pth2, DT_LIB_COLLECT_COL_VISIBLE, TRUE, -1);
+                               DT_LIB_COLLECT_COL_PATH, pth2, DT_LIB_COLLECT_COL_VISIBLE, TRUE,
+                               DT_LIB_COLLECT_COL_COUNT, (*(token + 1)?0:count), -1);
```

Comment:
Because I only know the count for the full path and the common path parts were already added to the tree before.
I could also recursively set the count for the parents of the current leaf, but then this function would be even longer. The callback is doing exactly the same and I think by splitting the code is easier to understand.